### PR TITLE
nts net train on cub200 2011

### PR DESCRIPTION
**Pull request by nicolalandro**
I add a new nts net model pretrained on cub200 2011

### CODE REVIEW

File: nicolalandro_ntsnet-cub200_ntsnet.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#     transforms.Normalize((0.485, 0.456, 0.406), (0.229, 0.224, 0.225)),
+# ])
+import torch
+model = torch.hub.load('nicolalandro/ntsnet-cub200', 'ntsnet', pretrained=True, **{'topN': 6, 'device':'cpu', 'num_classes': 200})
+# top_n_coordinates, concat_out, raw_logits, concat_logits, part_logits, top_n_index, top_n_prob = model(img)
+# if you need the output props use "concat_out"
+```
+
+### Model Description
+This is a nts-net pretrained with CUB200 2011 dataset. A fine grane dataset of birds species.
```
Comment (ailzhang) [inline nicolalandro_ntsnet-cub200_ntsnet.md:33]: nit: grane -> grained

File: nicolalandro_ntsnet-cub200_ntsnet.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+# transform_train = transforms.Compose([
+#     transforms.Resize((600, 600), Image.BILINEAR),
+#     transforms.CenterCrop((448, 448)),
+#     transforms.RandomHorizontalFlip(),  # solo se train
+#     transforms.ToTensor(),
+#     transforms.Normalize((0.485, 0.456, 0.406), (0.229, 0.224, 0.225)),
+# ])
+import torch
+model = torch.hub.load('nicolalandro/ntsnet-cub200', 'ntsnet', pretrained=True, **{'topN': 6, 'device':'cpu', 'num_classes': 200})
+# top_n_coordinates, concat_out, raw_logits, concat_logits, part_logits, top_n_index, top_n_prob = model(img)
```
Comment (ailzhang) [inline nicolalandro_ntsnet-cub200_ntsnet.md:28]: Would you mind adding an example about how the inference works on real bird picture?

File: nicolalandro_ntsnet-cub200_ntsnet.md (hunk 1 above)
Comment (nicolalandro) [reply nicolalandro_ntsnet-cub200_ntsnet.md:33]: Thanks! Scuse me for error.

File: nicolalandro_ntsnet-cub200_ntsnet.md (hunk 2 above)
Comment (nicolalandro) [reply nicolalandro_ntsnet-cub200_ntsnet.md:28]: I want to add also a better part of documentation in order to make easy to train it yourself. But now I do not have much time. The idea is to share it quickly with this small documentation and add some better explanation and sample in the future. Do you think it is possible?

**Comment by nicolalandro** [comment]
Hi, Is there any news? It is the first time I submit a pull request, maybe I should clik close?

File: nicolalandro_ntsnet-cub200_ntsnet.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+img = Image.open(urllib.request.urlopen(url))
+scaled_img = transform_test(img)
+torch_images = scaled_img.unsqueeze(0)
+
+with torch.no_grad():
+    top_n_coordinates, concat_out, raw_logits, concat_logits, part_logits, top_n_index, top_n_prob = model(torch_images)
+    
+    _, predict = torch.max(concat_logits, 1)
+    pred_id = predict.item()
+    print('bird class:', pred_id)
```
Comment (soumith) [inline nicolalandro_ntsnet-cub200_ntsnet.md:52]: it would be really helpful for the end user if they can know how they can get the bird name, not just the id.
Can you add that to this code snippet?

File: nicolalandro_ntsnet-cub200_ntsnet.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+### How to train
```
Comment (soumith) [inline nicolalandro_ntsnet-cub200_ntsnet.md:55]: this section is not really in line with the other Hub cards, and should just be removed.

The central idea of Hub is to get quick usage of trained models and get tangible outputs.
For referring to training mechanism, they will come to your github page and read more

File: nicolalandro_ntsnet-cub200_ntsnet.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+category: researchers
+title: ntsnet
+summary: a fine grane model for image classification.
```
Comment (soumith) [inline nicolalandro_ntsnet-cub200_ntsnet.md:7]: "a fine grane model for image classification." -> "a fine grained model for bird image classification"

File: nicolalandro_ntsnet-cub200_ntsnet.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+image: nts-net.png
+author: Moreno Carraffini and Nicola Landro
+tags: [vision]
+github-link: https://github.com/nicolalandro/ntsnet-cub200/archive/master.zip
+featured_image_1: no-image
```
Comment (soumith) [inline nicolalandro_ntsnet-cub200_ntsnet.md:12]: it would be good to add an image of a bird, maybe from the dataset. it is helpful for users when they browse the https://pytorch.org/hub/ page

File: nicolalandro_ntsnet-cub200_ntsnet.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+featured_image_2: no-image
+accelerator: "cuda-optional"
+---
+
+```python
+import torch
+model = torch.hub.load('nicolalandro/ntsnet-cub200', 'ntsnet', pretrained=True, **{'topN': 6, 'device':'cpu', 'num_classes': 200})
+```
+
+### How to use on bird
```
Comment (soumith) [inline nicolalandro_ntsnet-cub200_ntsnet.md:22]: "How to use on bird"  -> "Example"

File: nicolalandro_ntsnet-cub200_ntsnet.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    total_loss.backward()
+    raw_optimizer.step()
+    part_optimizer.step()
+    concat_optimizer.step()
+    partcls_optimizer.step()
+```
+
+
+### Model Description
+This is a nts-net pretrained with CUB200 2011 dataset. A fine graned dataset of birds species.
```
Comment (soumith) [inline nicolalandro_ntsnet-cub200_ntsnet.md:108]: "This is a" -> "This is an"

File: nicolalandro_ntsnet-cub200_ntsnet.md (hunk 4 above)
Comment (nicolalandro) [reply nicolalandro_ntsnet-cub200_ntsnet.md:55]: so is not import to train it from yourself? Is not an interesting case? I can see "pretrained=True" so you can also use "pretrained=False" and if you do not want to train it i do not know why exist this possibility. Other people do not specify it because train a resnet or something else is standard.

File: nicolalandro_ntsnet-cub200_ntsnet.md (hunk 3 above)
Comment (nicolalandro) [reply nicolalandro_ntsnet-cub200_ntsnet.md:52]: I'll do it!

File: nicolalandro_ntsnet-cub200_ntsnet.md (hunk 5 above)
Comment (nicolalandro) [reply nicolalandro_ntsnet-cub200_ntsnet.md:7]: Scuse me again, I will correct it!

File: nicolalandro_ntsnet-cub200_ntsnet.md (hunk 6 above)
Comment (nicolalandro) [reply nicolalandro_ntsnet-cub200_ntsnet.md:12]: Ok!

File: nicolalandro_ntsnet-cub200_ntsnet.md (hunk 8 above)
Comment (nicolalandro) [reply nicolalandro_ntsnet-cub200_ntsnet.md:108]: Ok!

File: nicolalandro_ntsnet-cub200_ntsnet.md (hunk 4 above)
Comment (soumith) [reply nicolalandro_ntsnet-cub200_ntsnet.md:55]: it's not that it's not important to train it yourself.
It's not important to put training mechanisms in the Hub cards, which are more for using model artifacts.
If people want to train, they will invariably come to your github repository to do so
